### DISCUSSION TITLE
OKD 4.7.0 installation problem on oVirt

### DISCUSSION THREAD

**Discussion by Gibadm2** [okd-project/okd#912] [General]
Hi,

Domain stend.local
Network 10.254.67.0/24
10.254.67.110 - VM for install OKD (okd.stend.local)
10.254.67.108 - IP reserved for API
10.254.67.109 - reserved for Ingress virtual IP
10.254.67.100 - vm.stend.local oVirt engine
10.254.67.101 - host1.stend.local
10.254.67.102 - host2.stend.local

File /etc/hosts:
<pre><code>
[root@okd ~]# cat /etc/hosts
10.254.67.100   vm vm.stend.local
10.254.67.101   host1 host1.stend.local
10.254.67.102   host2 host2.stend.local
10.254.67.110	okd okd.stend.local
10.254.67.108	api.okd-cluster api.okd-cluster.stend.local
10.254.67.109	*.apps.okd-cluster *.apps.okd-cluster.stend.local
10.254.67.111	dns dns.stend.local

127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
</code></pre>

On okd.stend.local:
<pre><code>[user@okd okd-4.7.0]$ ls | sort
kubectl
oc
openshift-client-linux-4.7.0-0.okd-2021-09-19-013247.tar.gz
openshift-install
openshift-install-linux-4.7.0-0.okd-2021-09-19-013247.tar.gz
README.md
$ sudo cp oc kubectl openshift-install /usr/local/bin/
$ cd ../cluster (empty directory)
$ openshift-install create cluster
Install version 4.8.0:
? Engine FQDN[:PORT] vm.stend.local
INFO Loaded the following PEM file:               
INFO 	Version: 3                                  
INFO 	Signature Algorithm: SHA256-RSA             
INFO 	Serial Number: 4096                         
INFO 	Issuer: CN=vm.stend.local.80657,O=stend.local,C=US 
INFO 	Validity:                                   
INFO 		Not Before: 2021-08-16 08:23:40 +0000 UTC  
INFO 		Not After: 2031-08-15 08:23:40 +0000 UTC   
INFO 	Subject: CN=vm.stend.local.80657,O=stend.local,C=US 
? Would you like to use the above certificate to connect to the Engine?  Yes
? Engine username admin@internal
? Engine password [? for help] ********
? Cluster Default
? Storage domain VMS
? Network ovirtmgmt
? Internal API virtual IP 10.254.67.108
? Ingress virtual IP 10.254.67.109
? Base Domain stend.local
? Cluster Name okd-cluster
? Pull Secret [? for help] *************************************
</code></pre>
Next installation progress:
<pre><code>INFO Obtaining RHCOS image file from 'https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/34.20210529.3.0/x86_64/fedora-coreos-34.20210529.3.0-openstack.x86_64.qcow2.xz?sha256=c86d0826300b403f528ea6f1bb81ac6ec9d1b73def80f3c9035730ea0bc5121d' 
INFO Creating infrastructure resources...         
INFO Waiting up to 20m0s for the Kubernetes API at https://api.okd-cluster.stend.local:6443... 
ERROR Attempted to gather ClusterOperator status after installation failure: listing ClusterOperator objects: Get "https://api.okd-cluster.stend.local:6443/apis/config.openshift.io/v1/clusteroperators": dial tcp 10.254.67.108:6443: connect: no route to host 
INFO Pulling debug logs from the bootstrap machine 
INFO Bootstrap gather logs captured here "/home/user/okd-4.7.0/log-bundle-20211005180316.tar.gz" 
FATAL Bootstrap failed to complete: failed waiting for Kubernetes API: Get "https://api.okd-cluster.stend.local:6443/version?timeout=32s": dial tcp 10.254.67.108:6443: connect: no route to host </code></pre>

**Comment by vrutkovs** [comment]
>FATAL Bootstrap failed to complete: failed waiting for Kubernetes API: Get "https://api.okd-cluster.stend.local:6443/version?timeout=32s": dial tcp 10.254.67.108:6443: connect: no route to host 

The machine which runs `openshift-install` can't find a route to LB

>INFO Pulling debug logs from the bootstrap machine 
>INFO Bootstrap gather logs captured here "/home/user/okd-4.7.0/log-bundle-20211005180316.tar.gz" 

Please attach [log bundle](https://docs.okd.io/latest/installing/installing-troubleshooting.html#installation-bootstrap-gather_installing-troubleshooting)

**Reply by Gibadm2** [reply]
Thanks. I'm uploaded logs to google drive [log-bundle-20211005180316.tar.gz](https://drive.google.com/file/d/1rMBInj6DBEkXxD1S59AI8BdVbb945NxY/view?usp=sharing)

**Reply by vrutkovs** [reply]
Right, this is invalid DHCP settings - even bootstrap host has `localhost` instead of a proper FQDN

**Reply by Gibadm2** [reply]
Can you tell me how I can fix this correctly?

**Reply by Gibadm2** [reply]
I'm create new network for OKD with DHCP and DNS servers.
DHCP config:
<pre><code>
subnet 192.168.0.0 netmask 255.255.255.0 {
  range 192.168.0.100 192.168.0.200;
  option domain-name-servers 192.168.0.1, 8.8.8.8;
  option domain-name "stend.local";
  option routers 192.168.0.1;
  option broadcast-address 192.168.0.255;
  default-lease-time 600;
  max-lease-time 7200;
}
</code></pre>
but installing okd cluster with error again
New logs [log-bundle-20211011163019.tar.gz](https://drive.google.com/file/d/1dpeBpCEtVX1Abbl67gsMDb-xC0SmOZnH/view?usp=sharing)

**Reply by JaimeMagiera** [reply]
Have you verified that your forward and reverse lookup is working for each IP? Have you verified that the nodes are successfully getting the info from the DHCP and DNS servers?

**Reply by Gibadm2** [reply]
> Have you verified that your forward and reverse lookup is working for each IP? Have you verified that the nodes are successfully getting the info from the DHCP and DNS servers?
Yes. The DHCP and DNS servers are working correctly
[openshift_install.log](https://drive.google.com/file/d/189VAuQzVAJytg1QrfL5dAlfiuqLUcyFb/view?usp=sharing)

<pre><code>
[root@okd ~]# nslookup api.okd-cluster.stend.local
Server:		192.168.0.20
Address:	192.168.0.20#53

Name:	api.okd-cluster.stend.local
Address: 192.168.0.11

[root@okd ~]# nslookup 192.168.0.11
11.0.168.192.in-addr.arpa	name = api.okd-cluster.stend.local.

[root@okd ~]# nslookup sasdasda.apps.okd-cluster.stend.local
Server:		192.168.0.20
Address:	192.168.0.20#53

Name:	sasdasda.apps.okd-cluster.stend.local
Address: 192.168.0.12

[root@okd ~]# nslookup 192.168.0.12
12.0.168.192.in-addr.arpa	name = *.apps.okd-cluster.stend.local.
<code></pre>

Installation progress:
<pre><code>
[user@okd cluster]$ openshift-install create cluster
? Platform ovirt
? oVirt cluster Default
? oVirt storage domain NFS
? oVirt network localnet
? Internal API virtual IP 192.168.0.11
? Internal DNS virtual IP 192.168.0.13
? Ingress virtual IP 192.168.0.12
? Base Domain stend.local
? Cluster Name okd-cluster
? Pull Secret [? for help] ******************************************
INFO Obtaining RHCOS image file from 'https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/32.20200629.3.0/x86_64/fedora-coreos-32.20200629.3.0-openstack.x86_64.qcow2.xz?sha256=dc2450ab1347f20eecc46f860105ddd4b62a677be6a5de648b31ef44bc114581' 
INFO The file was found in cache: /home/user/.cache/openshift-installer/image_cache/dd8863dfebe4cea2f6ad9ab84ff55f4a. Reusing... 
INFO Creating infrastructure resources...         
INFO Waiting up to 20m0s for the Kubernetes API at https://api.okd-cluster.stend.local:6443... 
INFO API v1.18.3 up                               
INFO Waiting up to 40m0s for bootstrapping to complete... 
E1019 17:40:23.748872    1107 reflector.go:307] k8s.io/client-go/tools/watch/informerwatcher.go:146: Failed to watch *v1.ConfigMap: Get https://api.okd-cluster.stend.local:6443/api/v1/namespaces/kube-system/configmaps?allowWatchBookmarks=true&fieldSelector=metadata.name%3Dbootstrap&resourceVersion=1&timeoutSeconds=447&watch=true: dial tcp 192.168.0.11:6443: connect: connection refused
...
igmaps?allowWatchBookmarks=true&fieldSelector=metadata.name%3Dbootstrap&resourceVersion=2382&timeoutSeconds=409&watch=true: dial tcp 192.168.0.11:6443: connect: connection refused
E1019 18:01:28.722906    1107 reflector.go:307] k8s.io/client-go/tools/watch/informerwatcher.go:146: Failed to watch *v1.ConfigMap: the server could not find the requested resource (get configmaps)
E1019 18:01:29.723982    1107 reflector.go:153] k8s.io/client-go/tools/watch/informerwatcher.go:146: Failed to list *v1.ConfigMap: the server could not find the requested resource (get configmaps)
E1019 18:01:30.725133    1107 reflector.go:153] k8s.io/client-go/tools/watch/informerwatcher.go:146: Failed to list *v1.ConfigMap: the server could not find the requested resource (get configmaps)
...
E1019 18:01:59.763558    1107 reflector.go:153] k8s.io/client-go/tools/watch/informerwatcher.go:146: Failed to list *v1.ConfigMap: Get https://api.okd-cluster.stend.local:6443/api/v1/namespaces/kube-system/configmaps?fieldSelector=metadata.name%3Dbootstrap&limit=500&resourceVersion=0: dial tcp 192.168.0.11:6443: connect: connection refused
INFO Pulling debug logs from the bootstrap machine 
ERROR Attempted to gather debug logs after installation failure: failed to create SSH client, ensure the private key is added to your authentication agent (ssh-agent) or specified with the --key parameter: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain 
FATAL Bootstrap failed to complete: failed to wait for bootstrapping to complete: timed out waiting for the condition 
[user@okd cluster]$ 
[user@okd cluster]$ openshift-install gather bootstrap --dir=.
INFO Pulling debug logs from the bootstrap machine 
FATAL failed to create SSH client, ensure the private key is added to your authentication agent (ssh-agent) or specified with the --key parameter: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain 
<code></pre>
